pkg/util/kubernetes/clustername/clustername.go

```diff
@@ -15,6 +17,10 @@ import (
 	"github.com/DataDog/datadog-agent/pkg/util/log"
 )
 
+var (
+	validClusterName = regexp.MustCompile(`^[a-z]([a-z0-9\-]{0,38}[a-z0-9])?$`)
```

Comment:
Could you add a comment there to say that this matches the naming rule in GKE? This way if the rule there changes, other contributors will understand that this line should be updated as well

Reply:
Documentation added.
With the link to GKE documentation as well as with a more human-friendly description of what the regular expression is supposed to match.